app/controllers/MovementsController.scala

```diff
@@ -20,87 +20,166 @@ import java.time.OffsetDateTime
+                            MessageSender(arrival.arrivalId, arrival.messages.head.messageCorrelationId)
+                          )
+                          .flatMap {
                             _ =>
-                              BadGateway
+                              arrival.messages.last.getState() match {
+                                case Some(state) =>
+                                  for {
+                                    _ <- arrivalMovementRepository.setMessageState(arrival.arrivalId,
+                                                                                   arrival.messages.length - 1,
+                                                                                   state.transition(SubmissionResult.Success))
+                                    _ <- arrivalMovementRepository.setState(arrival.arrivalId, arrival.state.transition(ArrivalSubmitted))
+                                  } yield {
+                                    Accepted("Message accepted")
+                                      .withHeaders("Location" -> routes.MovementsController.get(arrival.arrivalId).url)
+                                  }
+                                case None =>
+                                  Future.successful(InternalServerError)
+                              }
                           }
-                      }
+                          .recoverWith {
+                            case error =>
+                              logger.error(s"Call to EIS failed with the following Exception: ${error.getMessage}")
+
+                              arrival.messages.last.getState() match {
+                                case Some(state) =>
+                                  arrivalMovementRepository
+                                    .setMessageState(arrival.arrivalId, arrival.messages.length - 1, state.transition(SubmissionResult.Failure))
+                                    .map {
+                                      _ =>
+                                        BadGateway
+                                    }
+                                case None =>
+                                  Future.successful(BadGateway)
+                              }
+                          }
+                    }
+                }
+                .recover {
+                  case _ => {
+                    InternalServerError
+                  }
                 }
-            }
-            .recover {
-              case _ => {
-                // TODO: Add logging here so that we can be alerted to these issues.
-                InternalServerError
+          }
+      }
+
+  }
+
+  private def update_internal(arrival: Arrival, body: NodeSeq)(implicit hc: HeaderCarrier) =
+    arrivalMovementService.makeArrivalNotificationMessage(arrival.nextMessageCorrelationId)(body) match {
+      case None => Future.successful(BadRequest("Invalid data: missing either DatOfPreMES9, TimOfPreMES10 or DocNumHEA5"))
```

Comment:
Its worth revisiting this code for some refactor around common code [method update_internal and createMovement]